java/vortex-jni/src/main/java/dev/vortex/api/expressions/proto/TemporalMetadatas.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * (c) Copyright 2025 SpiralDB Inc. All rights reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.vortex.api.expressions.proto;
+
+import com.google.common.base.Preconditions;
+import java.io.ByteArrayOutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+@SuppressWarnings("MutablePublicArray")
```

Comment:
Another of my least favorite Java language choices... mutable final arrays

Reply:
I have a feature for you https://openjdk.org/jeps/502

Reply:
oh boy, maybe in 10 yrs java 25 will be the min java version for spark :D

Reply:
wait this is just suppliers.memoize

Reply:
that the runtime can trust to be immutable. The whole issue with existing final is that it was never trusted by the runtime

Reply:
Or you know `-XX:+UnlockExperimentalVMOptions -XX:+TrustFinalNonStaticFields` but apparently a lot of things break